Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -408,19 +451,53 @@ function New-LDAPSIdentitySource {
     else {
         Write-Host "No existing external identity sources found."
     }
-
     $Password = $Credential.GetNetworkCredential().Password
-    $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
+    $DestinationFileArray = @()
+    if ($PSBoundParameters.ContainsKey('SSLCertificatesSasUrl')) {
+        $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
+    } else {
+        $exportFolder = "$home/"
+        $remoteComputers = ,$PrimaryUrl
+        if ($PSBoundParameters.ContainsKey('SecondaryUrl')) { 
+            $remoteComputers += $SecondaryUrl 
+        }
+        
+        foreach ($computerUrl in $remoteComputers) {
+            try {
+                if (![uri]::IsWellFormedUriString($computerUrl, 'Absolute')) { throw }
+                $ParsedUrl = [System.Uri]$computerUrl
+            }
+            catch {
+                throw "Incorrect Url format entered from: $computerUrl"
+            }
+            if ($ParsedUrl.Host -match "^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$" -and [bool]($ParsedUrl.Host -as [ipaddress])) { 
+                throw "Incorrect Url format. $computerUrl is an IP address. Consider changing it to an URL"
```

Comment:
Since we're declaring it as invalid

>  Consider changing it to an URL

Should say something like "Use host name exactly as specified on the issued certificate".

Come to think, let's download the certificate and then comapre IssuedTo/CN to the address they are specifying,